edb/repl/utils.py

```diff
@@ -124,3 +124,17 @@ def normalize_name(name: str) -> str:
         return ''
 
     return name
+
+
+def get_filter_based_on_pattern(pattern: str,
+                                field: str = '.name',
+                                flag: str = '') -> str:
+    if flag:
+        flag = f'(?{flag})'
+
+    if pattern:
+        return f'FILTER re_test($${flag}{pattern}$$, {field})'
```

Comment:
`pattern` must be escaped. For example, your code will break if someone inputs `\l $$`. People can also do an EdgeQL injection this way. Make pattern a parameter of the query; pass flags as parameters.

Reply:
I've updated the code to use named query parameters and passing the string as a value to those.